docs/OSC-CONTEXT.md

```diff
@@ -0,0 +1,283 @@
+|    `OSC "11;…"` | "                                                          |
+|    `OSC "12;…"` | "                                                          |
+|    `OSC "13;…"` | "                                                          |
+|    `OSC "14;…"` | "                                                          |
+|    `OSC "15;…"` | "                                                          |
+|    `OSC "16;…"` | "                                                          |
+|    `OSC "17;…"` | "                                                          |
+|    `OSC "18;…"` | "                                                          |
+|    `OSC "19;…"` | "                                                          |
+|    `OSC "21;…"` | Query colors (kitty)                                       |
+|    `OSC "22;…"` | Cursor shape                                               |
+|    `OSC "46;…"` | Log file                                                   |
+|    `OSC "50;…"` | Set font                                                   |
+|    `OSC "51;…"` | Emacs shell                                                |
+|    `OSC "52;…"` | Manipulate selection data (aka clipboard)                  |
+|    `OSC "60;…"` | Query allowed                                              |
+|    `OSC "61;…"` | Query disallowed                                           |
+|    `OSC "99;…"` | Notifications (kitty)                                      |
+|   `OSC "104;…"` | Reset color                                                |
+|   `OSC "105;…"` | Enable/disable special color                               |
+|   `OSC "110;…"` | Reset colors                                               |
+|   `OSC "111;…"` | "                                                          |
+|   `OSC "112;…"` | "                                                          |
+|   `OSC "113;…"` | "                                                          |
+|   `OSC "114;…"` | "                                                          |
+|   `OSC "115;…"` | "                                                          |
+|   `OSC "116;…"` | "                                                          |
+|   `OSC "117;…"` | "                                                          |
+|   `OSC "118;…"` | "                                                          |
+|   `OSC "119;…"` | "                                                          |
+|   `OSC "133;…"` | Prompt/command begin/command end (finalterm/iterm2)        |
+|   `OSC "440;…"` | Audio (mintty)                                             |
+|   `OSC "633;…"` | vscode action (Windows Terminal)                           |
+|   `OSC "666;…"` | "termprop" (vte)                                           |
+|   `OSC "701;…"` | Locale (mintty)                                            |
+|   `OSC "777;…"` | Notification (rxvt)                                        |
```

Comment:
Fedora's GNOME Terminal and VTE had been using `OSC 777` as an extension point for a lot of things starting from January 2015.  :)

See:\
https://github.com/debarshiray/vte/commits/vte-0-64-cntnr-precmd-preexec-scroll/ \
https://github.com/debarshiray/gnome-terminal/commits/gnome-3-40-cntr-ntfy-autottl-ts/ 

Notifications (eg., *Command completed*) were the first.

Z shell style `precmd` and `preexec` notifications or *command begin* and *command end* to track the current foreground process.  On Bash these were emitted from `PS0` and `PROMPT_COMMAND`.  These were used to automatically update the terminal title and find out which command completed.  I believe we [played](https://lists.gnu.org/archive/html/bug-bash/2015-02/msg00014.html) a [role](https://lists.gnu.org/archive/html/bug-bash/2015-03/msg00003.html) in `PS0` becoming a thing.

Finally, to track the current container.

Reply:
As of Fedora 41 all the OSC `777` patches have been removed and we ship a stock upstream VTE.

Reply:
> ... and call it by the fancy name of termprops, and refer to the prior art as legacy.

I believe termprops is just generic key/value storage. But they added a bridge for the `777` stuff so I wouldn't need to support both the Fedora patches _and_ something else.